src/xenserver/opt/xensource/www/asset/fonts/icons.css

```diff
@@ -0,0 +1,58 @@
+@font-face {
+  font-family: "icons";
+  src: url("./icons.woff2?6fd33f90b90d0ac43e450f459069f7f3") format("woff2"), url("./icons.svg?6fd33f90b90d0ac43e450f459069f7f3#icons") format("svg");
+}
+.icon::before, [class^=icon-]::before, [class*=" icon-"]::before {
+  font-family: icons !important;
+  font-style: normal;
+  font-weight: normal !important;
+  font-variant: normal;
+  text-transform: none;
+  line-height: 1;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
+
+.icon.icon-browser::before {
+  content: "\f101";
+}
+
+.icon.icon-facebook::before {
+  content: "\f102";
+}
+
+.icon.icon-github::before {
+  content: "\f103";
+}
+
+.icon.icon-info-circle::before {
+  content: "\f104";
+}
+
+.icon.icon-spinner::before {
+  content: "\f105";
+}
+
+.icon.icon-terminal::before {
+  content: "\f106";
+}
+
+.icon.icon-twitter::before {
+  content: "\f107";
+}
+
+.icon.icon-user-headset::before {
+  content: "\f108";
+}
+
+.icon.icon-warn::before {
+  content: "\f109";
+}
+
+.icon.icon-wrench::before {
+  content: "\f10a";
+}
+
+.icon.icon-youtube::before {
+  content: "\f10b";
+}
```

Comment:
Please add an empty line at the EOF.

Reply:
yes, for your information this file is generated by fantasticon and a custom template
And why adding an empty line :/ ?

Reply:
It's not critical, but it's a convention: text files should end with a newline character. An well known AI explains it rather well:

![image](https://user-images.githubusercontent.com/451043/214823701-d038eefd-880f-455a-b5c6-3606e3877953.png)

This stackoverflow answer also covers it: https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

You can also see that git diff shows it when there's no newline at end of a file:
![image](https://user-images.githubusercontent.com/451043/214824635-4ee86816-9a97-4086-95f0-1088545cc2a8.png)

By the way, I checked the original jquery file (https://code.jquery.com/jquery-3.6.3.min.js), and it differs from yours by... One newline, at end of file. The upstream file should be copied as is, without any alterations.

Reply:
it's fixed by copying the real original file to the repo and commited it